### PR TITLE
Case multi select feature preview

### PR DESCRIPTION
**Pull request by benrudolph**
@snopoke @NoahCarnahan this supports the splitting of a column in a case export based on user options. i had to rework some of our abstraction to make it so `ExportItem`s didn't know how to split themselves. can be read 🐟 
screen shot 2016-04-23 at 11 51 20 am
screen shot 2016-04-23 at 11 49 58 am

one thing that doesn't work that well is if you create an export with the feature preview enabled, and then turn it off and then export it, it will still split the columns as if the flag was enabled since that's how it was created. similarly if you create an export with the flag disabled and then enable it, it won't let you add options; you'll have to create a new export. curious if you guys think this is an OK limitation. it will add significant messiness/complexity from what i can tell if we think that's an unreasonable expectation

cc: @millerdev

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/export/static/export/js/models.js, line 186 - Missing semicolon. (semi)
* corehq/apps/export/static/export/js/const.js, line 8 - 'Exports' is not defined. (no-undef)
* corehq/apps/export/static/export/js/const.js, line 10 - 'Exports' is not defined. (no-undef)
* corehq/apps/export/static/export/js/const.js, line 9 - 'Exports' is not defined. (no-undef)
* corehq/apps/export/static/export/js/const.js, line 6 - 'Exports' is not defined. (no-undef)
* corehq/apps/export/static/export/js/const.js, line 7 - 'Exports' is not defined. (no-undef)
* corehq/apps/export/static/export/js/models.js, line 190 - Missing semicolon. (semi)

File: corehq/apps/export/tests/test_export_conversion.py [hunk 1]
```diff
@@ -33,7 +34,13 @@
     CASE_HISTORY_TABLE,
 )
 
+MockRequest = namedtuple('MockRequest', 'domain')
 
+
+@mock.patch(
```
Comment (benrudolph) [inline corehq/apps/export/tests/test_export_conversion.py]: this is somewhat regrettable and messy, but couldn't figure out a clean way to get around it

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/export/static/export/js/models.js, line 186 - Missing semicolon. (semi)
* corehq/apps/export/static/export/js/const.js, line 8 - 'Exports' is not defined. (no-undef)
* corehq/apps/export/static/export/js/const.js, line 10 - 'Exports' is not defined. (no-undef)
* corehq/apps/export/static/export/js/const.js, line 9 - 'Exports' is not defined. (no-undef)
* corehq/apps/export/static/export/js/const.js, line 6 - 'Exports' is not defined. (no-undef)
* corehq/apps/export/static/export/js/const.js, line 7 - 'Exports' is not defined. (no-undef)
* corehq/apps/export/static/export/js/models.js, line 190 - Missing semicolon. (semi)

File: corehq/apps/export/static/export/js/const.js [hunk 2]
```diff
@@ -1,8 +1,15 @@
-/*
+/* globals Exports
+ *
```
Comment (dimagimon) [inline corehq/apps/export/static/export/js/const.js]: '*' is defined but never used (no-unused-vars)

Comment (benrudolph) [inline corehq/apps/export/static/export/js/const.js]: ?

**Comment by snopoke** [comment]
Not sure where this requirement is coming from but my original version of
multi select splitting had this feature and product vetoed it. Might want
to make sure this is a go before spending any more effort.

**Comment by benrudolph** [comment]
Huh interesting. I just saw that the current exports had the functionality
and thought we'd need to replicate it. Haha darn it! @dimagi/product any
thoughts?

**Comment by amsagoff** [comment]
@snopoke Here's what I recall: we put case multiselect splitting in Feature Preview, and productized form multiselect splitting. 

We did this because for cases, the user experience was complicated to get right, and the bigger pain point for users was on forms.  

When @benrudolph was working on new exports, he asked me if we needed to port over the multiselect splitting functionality. My thinking was that if significant people had the flag turned on, we should probably port it. It sounds like ben did port it over, but under the flag.

Ben, I would be curious to know how much has changed from the old version to this one (e.g. the "have to have flag on at time of making export" thing). And Simon, let me know if any of the backstory isn't how you remember it.

**Comment by benrudolph** [comment]
@amsagoff played around with the flag in the current state and it looks like it has the same limitations as the new version

**Comment by amsagoff** [comment]
cool, then good to go!

**Comment by benrudolph** [comment]
✅

**Comment by benrudolph** [comment]
i think this is ready to go @NoahCarnahan

**Pull request merged**